### ISSUE TITLE
Upgrade CVODE

### ISSUE THREAD

**Issue by spco** [AtChem/AtChem2#257]
LLNL have released SUNDIALS/CVODE 3.x : we should probably upgrade to use this unless there's a good reason not to.

**Comment by rs028** [comment]
I tried to update this, but it seems they changed the interface significantly. I would hold off the upgrade until the current work is finished.

**Comment by spco** [comment]
Yeah, I haven't looked into this at all - it may be some small changes required, or something more. We can take a look when we get round to it :)

**Comment by rs028** [comment]
Note: if CVODE is updated it overwrites the older version. That means that older models may not run anymore.

**Comment by spco** [comment]
The branch with this work on is https://github.com/spco/AtChem2/tree/upgrade_cvode , but I haven't looked at it since, so I can't say for sure what state it's in right now. It requires a number of modifications, and different function calls to previously.

**Comment by rs028** [comment]
Check the use of LAPACK (and BLAS?)
1. are they used? if not, should they be implemented?
2. if not used or necessary, drop from the list of requirements?

**Comment by rs028** [comment]
Bumping to higher priority because CVODE has now reached version 6.x (https://sundials.readthedocs.io/en/latest/cvode/Introduction_link.html#changes-from-previous-versions).
